modules/core/src/main/java/org/apache/ignite/internal/event/AbstractEventProducer.java

```diff
@@ -39,30 +41,24 @@ public abstract class AbstractEventProducer<T extends Event, P extends EventPara
     @Override
     public void listen(T evt, EventListener<? extends P> listener) {
         listenersByEvent.compute(evt, (evt0, listeners) -> {
-            List<EventListener<P>> newListeners;
-
-            if (listeners == null) {
-                newListeners = new ArrayList<>(1);
-            } else {
-                newListeners = new ArrayList<>(listeners.size() + 1);
-
-                newListeners.addAll(listeners);
+            if (nullOrEmpty(listeners)) {
```

Comment:
I suggest to never have empty lists in this map, so we can also dump the `isEmpty` check here

Reply:
fix it